VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java

```diff
@@ -2,32 +2,52 @@
 
 import android.content.res.Resources;
 import android.support.annotation.NonNull;
+import android.support.annotation.StringRes;
 import android.text.Html;
-import android.text.Spanned;
-import android.text.TextUtils;
 import android.text.format.DateUtils;
 
 import org.edx.mobile.R;
+import org.edx.mobile.util.ResourceUtil;
+
+import java.util.HashMap;
 
 public abstract class DiscussionTextUtils {
     private DiscussionTextUtils() {
     }
 
-    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
-        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
+    public static CharSequence getAuthorAttributionText(@NonNull DiscussionThread discussionThread, @NonNull Resources resources) {
```

Comment:
It's not a big deal, but most of the methods in Android's API take a
Context as a parameter even when they only require a Resources instance,
presumably for the convenience of the caller.

On Mon, Sep 21, 2015 at 9:53 PM, Brian Guertin <notifications@github.com>
wrote:

> In
> VideoLocker/src/main/java/org/edx/mobile/discussion/DiscussionTextUtils.java
> <https://github.com/edx/edx-app-android/pull/357#discussion_r39994829>:
>
> >
> >  public abstract class DiscussionTextUtils {
> >      private DiscussionTextUtils() {
> >      }
> >
> > -    public static String getAuthorAttributionText(@NonNull IAuthorData authorData, @NonNull Resources resources) {
> > -        String text = authorData.getAuthor() + " " + DateUtils.getRelativeTimeSpanString(
> > +    public static CharSequence getAuthorAttributionText(@NonNull DiscussionThread discussionThread, @NonNull Resources resources) {
>
> They don't need the whole context, they only need the Resources
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/357/files#r39994829>.
>